[DeviceMesh] Enable mesh universe concept in mesh comparison

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #165680


Since we now use the same _rank_map everywhere, we can just use it to differentiate between different mesh universe and get rid of root mesh comparison.


cc @H-Huang @awgu @wanchaol @fegin @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci

fduwjj committed:
[DeviceMesh] Enable mesh universe concept in mesh comparison

[ghstack-poisoned]

lw reviewed (approved):
Yes please!!! So happy we can finally do this! Go!!

fduwjj committed:
Update on "[DeviceMesh] Enable mesh universe concept in mesh comparison"



Since we now use the same _rank_map everywhere, we can just use it to differentiate between different mesh universe and get rid of root mesh comparison.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh] Enable mesh universe concept in mesh comparison"



Since we now use the same _rank_map everywhere, we can just use it to differentiate between different mesh universe and get rid of root mesh comparison.


cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]
